### PR TITLE
feat: add overlap analysis tool and fix CLI Node compatibility

### PR DESCRIPTION
**Pull request by joeycumines**
### Summary

This PR implements the `analyze overlaps` tool to detect visual overlaps, layout overflows, and overlay patterns. It also updates CLI file operations to use standard Node libraries instead of Bun APIs so the published package runs correctly in Node environments.

### What changed

- Add `analyze overlaps` CLI command, RPC command, and ToolDef.
- Implement world-matrix visual bounds computation to account for nested rotations and ancestor clipping.
- Replace `Bun.write` and `Bun.file` with `node:fs/promises` in the `convert`, `eval`, and `export` CLI commands.
- Add `@open-pencil/core/package.json` to subpath exports.

### Validation

- [x] `bun run check`
- [x] Tests added or updated, or not needed because: Added test suites for overlap CLI, bounds computation, page scoping, and core overlap logic. Added workspace build requirements for CLI integration tests.
- [x] CHANGELOG.md updated, or not needed because: Documented new overlap features and CLI Node compatibility fixes.

## Summary by CodeRabbit

* **New Features**
  * Added `analyze overlaps` to detect visual overlaps (sibling overlaps, parent overflow, overlays) with scope, category, severity, page selection, and minimum area/ratio filters, plus optional `--json` output and limiting.
  * Added overlap-analysis automation exports for programmatic use.
* **Bug Fixes**
  * Improved overlap/bounds accuracy, especially for rotated clipping chains and stroke/text-derived bounds.
  * Handled `limit` edge cases (0/negative) and trimmed whitespace for scope/severity inputs.
  * Reduced CI flakiness with increased timeouts for slow fixture tests.
* **Documentation**
  * Updated README/changelog examples to include overlaps analysis.
* **Tests**
  * Added CLI and overlap-analysis test coverage, including rotated/clipping bounds scenarios.

### CODE REVIEW

File: packages/core/src/tools/analyze.ts [hunk 1]
```diff
@@ -3,5 +3,14 @@ export { analyzeColors } from './analyze/colors'
...
 export { wrapEvalCode } from './analyze/eval/wrap'
+export { analyzeOverlaps, computeOverlaps } from './analyze/overlaps'
+export type {
+  AnalyzeOverlapsArgs,
+  AnalyzeOverlapsResult,
+  OverlapCategory,
+  OverlapItem,
+  OverlapNodeSummary,
+  OverlapSeverity
+} from './analyze/overlaps'
```
[the remark this answers is not in the document]
Comment (joeycumines) [reply packages/core/src/tools/analyze.ts:17]: Fixed in ad22513b. Added `AnalyzeOverlapsSummary`, `OverlapIntersection`, and `OverlapScope` to the type export in `packages/core/src/tools/analyze.ts`, so the full overlap type surface is reachable from the barrel without a deep import.

File: packages/core/src/tools/analyze/overlaps/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+    { x: baseCorners[6], y: baseCorners[7] }
+  ])
+
+  const stroke = strokeOverflow(node.strokes)
+  if (stroke > 0) {
+    bounds.minX -= stroke
+    bounds.minY -= stroke
+    bounds.maxX += stroke
+    bounds.maxY += stroke
+  }
```
[the remark this answers is not in the document]
Comment (joeycumines) [reply packages/core/src/tools/analyze/overlaps/helpers.ts]: Fixed in ad22513b. `computeNodeVisualBounds` now expands the local rectangle by `strokeOverflow` before mapping the corners through the world matrix, so a rotated stroked node gets its true rotated footprint (`stroke*(|cosθ|+|sinθ|)` per side) instead of the underestimated uniform canvas-space expansion. Added a test (`a stroked rotated node expands bounds along the rotated axes`) that places a target in the gap between the old naive expansion and the rotation-aware bounds, and confirms the unstroked node does not reach it.

File: packages/core/src/tools/analyze/overlaps/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+  graph: SceneGraph
+): { bounds: VisualBounds; area: number } {
+  let bounds = computeNodeVisualBounds(node, graph)
+  const clips = collectClipChain(graph, node)
+  for (const clip of clips) {
+    const clipped = clipBoundsToPolygon(bounds, clip)
+    if (!clipped) {
+      return { bounds: EMPTY_BOUNDS, area: 0 }
+    }
+    bounds = clipped
```
[the remark this answers is not in the document]
Comment (joeycumines) [reply packages/core/src/tools/analyze/overlaps/helpers.ts]: Fixed in ad22513b. Added `clipPolygon` to `geometry.ts` and rewrote `computeNodeBounds` to preserve the clipped polygon through the full clip chain, collapsing to an AABB only at the end. `clipBoundsToPolygon` now delegates to `clipPolygon`. Added `tests/engine/geometry/clip-polygon.test.ts` with a chained-clip case proving a corner removed by an inner diamond clip stays excluded from a later corner-covering square clip.

File: packages/core/src/tools/analyze/overlaps/index.ts [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+  ]
+
+  const sorted = orderBy(
+    overlaps,
+    [(o) => scoredSeverity(o.severity), (o) => o.area],
+    ['desc', 'desc']
+  )
+
+  const limit = Math.max(0, Number.isFinite(Number(args.limit)) ? Number(args.limit) : 100)
+  const trimmed = limit > 0 ? sorted.slice(0, limit) : sorted
```
[the remark this answers is not in the document]
Comment (joeycumines) [reply packages/core/src/tools/analyze/overlaps/index.ts]: Fixed in ad22513b. A `limit` of 0 (or a negative value clamped to 0) now caps the returned `overlaps` to an empty list via `sorted.slice(0, limit)`; the summary (`overlapCount`, `byCategory`, `bySeverity`) still reflects the full `sorted` set. Added tests for `limit: 0` and a negative limit.

File: packages/core/src/tools/analyze/overlaps/params.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  'critical',
+  'major',
+  'minor',
+  'info'
+]
+
+export function parseOverlapScope(raw: string | undefined): OverlapScope | undefined {
+  if (!raw) return undefined
+  const normalized = raw.toLowerCase()
+  return VALID_OVERLAP_SCOPES.find((scope) => scope === normalized)
```
[the remark this answers is not in the document]
Comment (joeycumines) [reply packages/core/src/tools/analyze/overlaps/params.ts:28]: Fixed in ad22513b. `parseOverlapScope` and `parseOverlapSeverity` now `trim()` before lowercasing, matching `parseOverlapCategories`. Added a test asserting `"  same-parent  "` and `"  minor  "` resolve.

**Comment by joeycumines** [comment]
All five actionable CodeRabbit findings are addressed in ad22513b and replies are posted on each thread. Summary:

- **Barrel type exports** (`analyze.ts`): added `AnalyzeOverlapsSummary`, `OverlapIntersection`, `OverlapScope`.
- **Rotated stroke bounds** (`helpers.ts`): expand the local rectangle by `strokeOverflow` before the world-matrix transform, so rotated stroked nodes get their true footprint.
- **Clip chain precision** (`helpers.ts` + `geometry.ts`): new `clipPolygon` keeps the clipped polygon through the full ancestor chain instead of collapsing to an AABB between clips; `clipBoundsToPolygon` now delegates to it.
- **limit=0 / negative limit** (`index.ts`): `sorted.slice(0, limit)` caps the returned overlaps to an empty list; summary still reflects the full set.
- **Trim scope/severity** (`params.ts`): `trim()` added to `parseOverlapScope` and `parseOverlapSeverity`.

Each fix has a covering test (overlap bounds/index tests + a new `tests/engine/geometry/clip-polygon.test.ts`). `bun run check`, `bun run test:dupes`, and the overlap/geometry unit suites are green locally.

@coderabbitai please re-review.

File: tests/engine/tools/analyze/overlaps/bounds.test.ts [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+      x: 45,
+      y: 45,
+      width: 10,
+      height: 10,
+      rotation: 45
+    })
+    rect(plain, 'GapTarget', plainPage, 18, 48, 2, 2)
+    const plainResult = computeOverlaps(plain, { category: 'sibling-overlap' })
+    expect(plainResult.summary.overlapCount).toBe(0)
+  })
```
[the remark this answers is not in the document]
Comment (joeycumines) [reply tests/engine/tools/analyze/overlaps/bounds.test.ts:236]: Fixed in 99e4153e. The `RotatedPlain` control node is now created on `plain` (the unstroked `SceneGraph`) instead of `graph`, so the control assertion actually validates the same setup without a stroke.
